File: Semester 1/Python Lab/Practice/Palindrome Number.py
def palindrome(n):
    temp = n
    reverse = 0
    while temp>0:
        reverse = reverse*10 + temp%10
        temp //= 10
    return reverse == n

number = int(input("Enter a number: "))
if(palindrome(number)):
    print(f"{number} is a palindrome number")
else:
    print(f"{number} is not a palindrome number")

File: Semester 3/CC Lab/palindromeNumber.java
import java.util.Scanner;

public class palindromeNumber
{
    boolean isPalindrome(long number)
    {
        long numCopy = number;
        long checkNum = 0;
        while(number>0)
        {
            checkNum =checkNum*10 + number %10;
            number/=10;
        }
        return (checkNum == numCopy);
    }
    public static void main(String[] args)
    {
        Scanner keyboard = new Scanner(System.in);
        palindromeNumber check = new palindromeNumber();
        System.out.print("Enter a number: ");
        long number = keyboard.nextLong();
        if(check.isPalindrome(number))
        {
            System.out.println("The number is palindrome.");
        }
        else
        {
            System.out.println("The number is not palindrome.");
        }
        System.out.println();
        keyboard.close();
    
    }
}